src/registry/customerProfile/digitalAddresses.ts

```diff
@@ -0,0 +1,69 @@
+import type {
+  CustomerProfileData,
+  CustomerProfileProperties,
+  DigitalAddress,
+  DigitalAddressType,
+} from '@open-formulieren/types';
+import {useFormikContext} from 'formik';
+import {useState} from 'react';
+import {useAsync} from 'react-use';
+
+import {useFormSettings} from '@/hooks';
+
+import type {DigitalAddressGroup, DigitalAddressesResponseBody} from './types';
+
+interface UseDigitalAddresses {
+  digitalAddresses: DigitalAddressesResponseBody;
+}
+
+export const useDigitalAddresses = (
+  profileComponentKey: string,
+  digitalAddressTypes: CustomerProfileProperties['digitalAddressTypes']
+): UseDigitalAddresses => {
+  const {setFieldValue} = useFormikContext<CustomerProfileData>();
+  const formSettings = useFormSettings();
+  const [digitalAddresses, setDigitalAddresses] = useState<DigitalAddressesResponseBody>([]);
+
+  if (!formSettings?.componentParameters?.customerProfile) {
+    throw new Error('Customer profile component parameters not configured');
+  }
+
+  const {fetchDigitalAddresses} = formSettings.componentParameters.customerProfile;
+
+  useAsync(async () => {
+    const result = await fetchDigitalAddresses(profileComponentKey);
+    if (!result) {
+      // Initial the form with empty values if fetch failed.
+      digitalAddressTypes.forEach((type, index) => {
+        setFieldValue(`${profileComponentKey}.${index}`, {
+          address: '',
+          type: type,
+          useOnlyOnce: true,
+        } satisfies DigitalAddress);
+      });
+      setDigitalAddresses([]);
```

Comment:
you can also just make this the return state of the `useAsync` hook, it already uses `useState` under the hood so you're basically duplicating the implementation and using only half of the hook :)

```ts
const {loading, value, error} = useAsync(...);
// value contains the return value from the wrapped callback
```